### PR TITLE
Im 20200615 mysql migration syntax

### PR DESCRIPTION
**Pull request by IanMMarshall**
Fix/enable migration tasks for MySQL. Ensured that these now work for both MySQL 5.7 and 8.0.

### CODE REVIEW

File: hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/RenameColumnTask.java [hunk 1]
```diff
@@ -82,6 +84,20 @@ public void doExecute() throws SQLException {
 					throw new SQLException("Can not rename " + getTableName() + "." + myOldName + " to " + myNewName + " because both columns exist and data exists in " + myNewName);
 				}
 
+				if (getDriverType().equals(DriverTypeEnum.MYSQL_5_7)) {
+					// Some DBs such as MYSQL require that foreign keys depending on the column be dropped before the column itself is dropped.
```
Comment (fil512) [inline hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/RenameColumnTask.java:88]: In the rename scenario, don't we need to add those constraints back in after the rename?

File: hapi-fhir-jpaserver-migrate/src/test/java/ca/uhn/fhir/jpa/migrate/taskdef/DropColumnTest.java [hunk 2]
```diff
@@ -34,4 +36,39 @@ public void testDropColumn() throws SQLException {
 
 	}
 
+	@Test
+	public void testDropForeignKeyColumn() throws SQLException {
```
Comment (fil512) [inline hapi-fhir-jpaserver-migrate/src/test/java/ca/uhn/fhir/jpa/migrate/taskdef/DropColumnTest.java:40]: I recommend adding a test for the rename column scenario to validate we don't lose the fk constraints on rename

File: hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/RenameColumnTask.java (hunk 1 above)
Comment (IanMMarshall) [reply hapi-fhir-jpaserver-migrate/src/main/java/ca/uhn/fhir/jpa/migrate/taskdef/RenameColumnTask.java:88]: Good question. In this case, no I do not think there is a need to add the constraint back after rename. The constraint being dropped in this case depends on a column that is also being dropped (an existing column being replaced by renaming another existing column). As the renamed column would potentially have different data values, there is no guarantee that the old constraint will still be relevant.  

For non-MySQL databases, it appears that these types of constraints are dropped automatically when the column is dropped. This does not happen for MySQL and so the change ensures that the constraint is similarly dropped in the case of MySQL.

File: hapi-fhir-jpaserver-migrate/src/test/java/ca/uhn/fhir/jpa/migrate/taskdef/DropColumnTest.java (hunk 2 above)
Comment (IanMMarshall) [reply hapi-fhir-jpaserver-migrate/src/test/java/ca/uhn/fhir/jpa/migrate/taskdef/DropColumnTest.java:40]: Good call. I have added tests for various scenarios involving renaming of columns that are used in foreign key constraints to confirm the existing behaviour and to ensure that the behaviour for MySQL use cases effectively matches that for non-MySQL DBs.

**Pull request merged**